js/bitmex.js

```diff
@@ -2666,6 +2697,46 @@ module.exports = class bitmex extends Exchange {
         return await this.privatePostPositionIsolate (this.extend (request, params));
     }
 
+    async fetchDepositAddress (code, params = {}) {
+        /**
+         * @method
+         * @name bitmex#fetchDepositAddress
+         * @description fetch the deposit address for a currency associated with this account
+         * @param {string} code unified currency code
+         * @param {object} params extra parameters specific to the bitmex api endpoint
+         * @param {network} params.network deposit chain, can view all chains via this.publicGetWalletAssets, default is eth, unless the currency has a default chain within this.options['networks']
+         * @returns {object} an [address structure]{@link https://docs.ccxt.com/en/latest/manual.html#address-structure}
+         */
+        const networkCode = this.safeStringUpper (params, 'network');
+        const networks = this.safeValue (this.options, 'networks', {});
+        let networkId = undefined;
+        if (networkCode === undefined) {
+            networkId = this.safeStringLower (networks, code, 'eth');
```

Comment:
this is very error-prone, why not require the network in `params` explicitly? 

For instance, `bitmex.fetchDepositAddress (USDT)` fails with `bitmex {"error":{"message":"Invalid network for currency","name":"HTTPError"}}`

Reply:
it just feels kinda unnecessary to require it, 80% of the time it's ETH, then 19% of the time, it's the same as the currency code

Reply:
@samgermain I would not be so sure about it, most people use the network with the lower fees, and as I said, something so simple as `bitmex.fetchDepositAddress (USDT)` fails, so I think we should change it